### PR TITLE
Development

### PR DESCRIPTION
**Pull request by Louquinze**
renamed the hp therefore all *csv and *json need to be changed

also redo the example

### CODE REVIEW

File: examples/40_advanced/example_text_preprocessing.py [hunk 1]
```diff
@@ -1,79 +1,98 @@
 # -*- encoding: utf-8 -*-
 """
-==================
-Text Preprocessing
-==================
-This example shows, how to use text features in *auto-sklearn*. *auto-sklearn* can automatically
-encode text features if they are provided as string type in a pandas dataframe.
+==============
```
Comment (mfeurer) [inline examples/40_advanced/example_text_preprocessing.py]:
```suggestion
==================
```

File: examples/40_advanced/example_text_preprocessing.py [hunk 2]
```diff
@@ -1,79 +1,98 @@
...
+Text preprocessing
+==============
```
Comment (mfeurer) [inline examples/40_advanced/example_text_preprocessing.py]:
```suggestion
==================
```

File: examples/40_advanced/example_text_preprocessing.py [hunk 3]
```diff
@@ -1,79 +1,98 @@
...
 
 ############################################################################
 # Data Loading
 # ============
 
-X, y = sklearn.datasets.fetch_openml(data_id=40945, return_X_y=True)
-
-# by default, the columns which should be strings are not formatted as such
-print(f"{X.info()}\n")
+newsgroups_train = fetch_20newsgroups(subset="train")
```
Comment (mfeurer) [inline examples/40_advanced/example_text_preprocessing.py]: Would it make sense to use the `remove` argument here to remove columns that make it too easy?

File: examples/40_advanced/example_text_preprocessing.py [hunk 4]
```diff
@@ -1,79 +1,98 @@
...
-    # Bellow two flags are provided to speed up calculations
-    # Not recommended for a real implementation
-    initial_configurations_via_metalearning=0,
-    smac_scenario_args={"runcount_limit": 1},
+############################################################################
+# Build and fit a classifier
+# ==========================
+
+automl = autosklearn.classification.AutoSklearnClassifier(
+    # set the time high enough text preprocessing can create many new features
```
Comment (mfeurer) [inline examples/40_advanced/example_text_preprocessing.py]: Does 20 newsgroup work in the setting on the left? That would be preferable for running this example in the github actions.

Comment (mfeurer) [reply examples/40_advanced/example_text_preprocessing.py]: Maybe we should also use a smaller dataset? You can use the following script to scan on OpenML for datasets containing string data:
```
import openml

datasets = openml.datasets.list_datasets()
for did in datasets:
    try:
        dataset = openml.datasets.get_dataset(did, download_data=False, download_qualities=False)
        for feat in dataset.features:
            if dataset.features[feat].data_type == 'string':
                print(did)
                break
    except Exception as e:
        print(e)
        continue
```

**Review by mfeurer — COMMENTED** [review]
@eddiebergman any idea how to fix PEP8 here?

File: examples/40_advanced/example_text_preprocessing.py (hunk 2 above)
Comment (eddiebergman) [inline examples/40_advanced/example_text_preprocessing.py]:
```suggestion
==================
```

File: examples/40_advanced/example_text_preprocessing.py (hunk 1 above)
Comment (eddiebergman) [inline examples/40_advanced/example_text_preprocessing.py]:
```suggestion
==================
```

**Comment by eddiebergman** [comment]
The line to long errors in pre-commit can be fixed by adding `# noqa: E501` to the end of those lines. I was rethinking that perhaps 100 line length is fine but that's a seperate thing to discuss, it wouldn't prevent these errors anyway.

The other solution is to have modules import in the `__init__` so the imports aren't this long.

File: examples/40_advanced/example_text_preprocessing.py (hunk 3 above)
Comment (Louquinze) [reply examples/40_advanced/example_text_preprocessing.py]: Do not know what argument i to remove, because 20 Newsgroups has only 1 column

File: examples/40_advanced/example_text_preprocessing.py (hunk 4 above)
Comment (Louquinze) [reply examples/40_advanced/example_text_preprocessing.py]: the example yields ~80% acc. on the test set. Selecting random would be  5% for 20 labels. Therefore i would say that the example works. But it also runs 300 sec. which are 5 min. So if that is to long i can search another dataset.

Comment (mfeurer) [reply examples/40_advanced/example_text_preprocessing.py:72]: Sorry, I meant, would the example work when you restrict it to use only a single configuration?

**Comment by mfeurer** [comment]
The doc build failure appeared to be unrelated, so I just restarted it. However, the `pre-commit` fails right now, could you please have a look into this?

**Comment by eddiebergman** [comment]
> The doc build failure appeared to be unrelated, so I just restarted it. However, the `pre-commit` fails right now, could you please have a look into this?

`make format`

**Comment by Louquinze** [comment]
> > The doc build failure appeared to be unrelated, so I just restarted it. However, the `pre-commit` fails right now, could you please have a look into this?
> 
> `make format` for the formatting. This bug with the leadboard as shown in the docs is triggered by this example. While not directly related to the example, it often occurs when no models are found as the id's of models get messed up.

i did make format pre-commit works now

File: examples/40_advanced/example_text_preprocessing.py [hunk 5]
```diff
@@ -80,13 +80,13 @@
 # View the models found by auto-sklearn
 # =====================================
 
-print(automl.leaderboard())
+# print(automl.leaderboard())
```
Comment (mfeurer) [inline examples/40_advanced/example_text_preprocessing.py:83]: Hey, why do you add a comment here?

File: examples/40_advanced/example_text_preprocessing.py [hunk 6]
```diff
@@ -80,13 +80,13 @@
...
 
 ############################################################################
 # Print the final ensemble constructed by auto-sklearn
 # ====================================================
 
-pprint(automl.show_models(), indent=4)
+# pprint(automl.show_models(), indent=4)
```
Comment (mfeurer) [inline examples/40_advanced/example_text_preprocessing.py:89]: And here?

File: examples/40_advanced/example_text_preprocessing.py (hunk 5 above)
Comment (Louquinze) [reply examples/40_advanced/example_text_preprocessing.py:83]: the print function somehow yields an error in the auto check on git so i commented it out but i can include it again, but then it probably will raise an error

File: examples/40_advanced/example_text_preprocessing.py (hunk 4 above)
Comment (Louquinze) [reply examples/40_advanced/example_text_preprocessing.py:72]: is there a parameter for setting autosklearn to it or is that max_time == timer per model ?

File: examples/40_advanced/example_text_preprocessing.py (hunk 5 above)
Comment (eddiebergman) [reply examples/40_advanced/example_text_preprocessing.py]: These errors came up before and I'm not certain on the reason that it happens.

* https://github.com/automl/auto-sklearn/blob/master/autosklearn/estimators.py#L839

Essentially there is an `id` in the ensemble that is not included in the run history of SMAC. As it's model id 1, my best guess is that it is to do with the dummy model not being run by SMAC in conjucntion with some other outcome state.

My best guess is that no model found was better than the Dummy model.

File: examples/40_advanced/example_text_preprocessing.py (hunk 4 above)
Comment (eddiebergman) [reply examples/40_advanced/example_text_preprocessing.py:72]: I would read through the entire API and manual now that you have a bit more familiarity, to know what's possible and what's not
https://automl.github.io/auto-sklearn/master/api.html

Comment (mfeurer) [reply examples/40_advanced/example_text_preprocessing.py:80]: It has been there in the previous version of the example: `smac_scenario_args={"runcount_limit": 1}`

**Review by mfeurer — CHANGES_REQUESTED** [review]
That looks good now. I believe the example could be drastically simplified by restricting the categories via the argument `categories` to the [function load 20 newsgroups](https://scikit-learn.org/stable/modules/generated/sklearn.datasets.fetch_20newsgroups.html).

File: examples/40_advanced/example_text_preprocessing.py [hunk 7]
```diff
@@ -22,67 +22,54 @@
...
-# explicitly label text column as string
-X_test = df_test.astype({"X": "string", "y": "category"})
-X_test = X_test[~X_test["y"].isin(five_newsgroups_labels)]
-y_test = X_test.pop("y")
+############################################################################
+# Creating a pandas dataframe
+# ===========================
+# Both categorical and text features are often strings. Python Pandas stores python stings
+# in the generic `object` type. Please ensure that the correct
+# `dtype <https://pandas.pydata.org/docs/user_guide/basics.html#dtypes>` is applied to the correct
```
Comment (mfeurer) [inline examples/40_advanced/example_text_preprocessing.py:45]:
```suggestion
# `dtype <https://pandas.pydata.org/docs/user_guide/basics.html#dtypes>_` is applied to the correct
```

Comment (mfeurer) [reply examples/40_advanced/example_text_preprocessing.py:45]: Or must the underscore go after the backtick :thinking:

File: examples/40_advanced/example_text_preprocessing.py [hunk 8]
```diff
@@ -22,67 +22,54 @@
...
 ############################################################################
 # Build and fit a classifier
 # ==========================
 
+# create an autosklearn Classifier or Regressor depending on your task at hand.
 automl = autosklearn.classification.AutoSklearnClassifier(
-    # set the time high enough text preprocessing can create many new features
-    time_left_for_this_task=300,
-    per_run_time_limit=30,
+    time_left_for_this_task=60,  # absolute time limit for fitting the ensemble
```
Comment (mfeurer) [inline examples/40_advanced/example_text_preprocessing.py:60]:
```suggestion
    time_left_for_this_task=60,
```

File: examples/40_advanced/example_text_preprocessing.py [hunk 9]
```diff
@@ -22,67 +22,54 @@
...
+    per_run_time_limit=30,  # time limit for single models (ensures seeing a variety of models)
```
Comment (mfeurer) [inline examples/40_advanced/example_text_preprocessing.py:61]:
```suggestion
    per_run_time_limit=30,
```

File: examples/40_advanced/example_text_preprocessing.py [hunk 10]
```diff
@@ -22,67 +22,54 @@
...
     tmp_folder="/tmp/autosklearn_text_example_tmp",
 )
-automl.fit(X_train, y_train, dataset_name="20_Newsgroups")
+
+automl.fit(  # fit our model to the training data
+    X=X_train,  # passing training data (encoded as pandas dataframe)
+    y=y_train,  # passing training labels
+    # ('array like' object: pandas Series, numpy array, python list etc.)
```
Comment (mfeurer) [inline examples/40_advanced/example_text_preprocessing.py:68]: I don't fully understand why this comment is here.

File: examples/40_advanced/example_text_preprocessing.py [hunk 11]
```diff
@@ -100,5 +87,7 @@
 # Get the Score of the final ensemble
 # ===================================
 
+# get predictions for formerly unseen data. Ensure that the data has the same format as the training
```
Comment (mfeurer) [inline examples/40_advanced/example_text_preprocessing.py:90]: I don't fully understand why this comment is here.

File: examples/40_advanced/example_text_preprocessing.py (hunk 10 above)
Comment (Louquinze) [reply examples/40_advanced/example_text_preprocessing.py:68]: i will minimize it

File: examples/40_advanced/example_text_preprocessing.py (hunk 7 above)
Comment (Louquinze) [reply examples/40_advanced/example_text_preprocessing.py:45]: example_debug_logging.py makes it like this
``SMAC documentation <https://github.com/automl/SMAC3>`_`

File: examples/40_advanced/example_text_preprocessing.py (hunk 11 above)
Comment (Louquinze) [reply examples/40_advanced/example_text_preprocessing.py:90]: i think i overthought this i think it is somehow self explanatory in the way the variables are named

**Review by mfeurer — APPROVED** [review]
